example_codes/controller_vs_agent.py

```diff
@@ -96,18 +113,19 @@ def act(self, observation, reward, done):
 
 pushed_keys = ["0","0","0","0","0","0","0","0","0","0","0","0"]
 
-state = env.reset()
+env.reset()
 env.render()
-#env.viewer.window.on_key_press = get_keys(pushed_keys)
-#env.viewer.window.on_key_release = get_keys(pushed_keys)
 agent = Agent(env.action_space)
 action = f'{0|2048:012b}'
-ob, reward, done, _ = env.step( action + "".join(pushed_keys) )
-while True:
+ob, reward, done, info = env.step( action + "".join(pushed_keys) )
+while(True):
     for i in range(100000):
         pushed_keys = get_keys(pushed_keys)
         action = agent.act(ob, reward, done)
-        ob, reward, done, _ = env.step("".join(pushed_keys) + action )
+        ob, reward, done, info = env.step("".join(pushed_keys) + action )
         env.render()
         sleep(1/90)
+        if(info['wins'] == 1 or info['enemy_rounds_won'] == 1 or info['rounds_won'] == 1 or done or i > 4000):

```

Comment:
This should be calculated by the environment (by changing how `done` is set) but that's just my opinion
(Also parentheses are so... javascript :D)

Reply:
For now it does the job :D,